hypothesis-python/src/hypothesis/internal/reflection.py

```diff
@@ -536,3 +536,14 @@ def accept(proxy):
         return impersonate(target)(wraps(target)(define_function_signature(
             target.__name__, target.__doc__, getfullargspec(target))(proxy)))
     return accept
+
+
+def is_running_in_repl():
+    """Tries to guess if Hypothesis is running from inside an interactive
+    REPL or inside a script."""
+    # The last frame in this list is the outermost call on the stack.
+    # In this case, the filename will be '<stdin>' if we're running inside
+    # the REPL.
+    stack = inspect.stack()
```

Comment:
I think `hasattr(sys, 'ps1')` works for detecting if you're in the repl.

Reply:
That seems like a better approach, and mostly exposes that how I thought I was testing the REPL actually wasn't. :'(